namerd/iface/destination/src/main/scala/io/buoyant/namerd/iface/destination/DestinationService.scala

```diff
@@ -0,0 +1,150 @@
+package io.buoyant.namerd.iface.destination
+
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.{Address, _}
+import com.twitter.io.Buf
+import com.twitter.logging.Logger
+import com.twitter.util._
+import io.buoyant.grpc.runtime.{GrpcStatus, Stream}
+import io.linkerd.proxy.destination.Destination
+import io.linkerd.proxy.destination.Update.OneofUpdate
+import io.linkerd.proxy.destination.Update.OneofUpdate.{NoEndpoints => EndpointsNone, _}
+import io.linkerd.proxy.destination._
+import io.linkerd.proxy.net.IPAddress.OneofIp
+import io.linkerd.proxy.net.{IPAddress, TcpAddress}
+import java.net.{InetAddress, InetSocketAddress}
+
+class DestinationService(
+  interpreter: NameInterpreter
+) extends Destination {
+  import DestinationService._
+
+  //Diff implementation for Address Set
+  class AddressDiffable extends Diffable[Set] {
+    override def diff[T](
+      left: Set[T],
+      right: Set[T]
+    ): Diff[Set, T] = AddressDiff(right -- left, left -- right)
+
+    override def empty[T]: Set[T] = Set.empty
+  }
+
+  case class AddressDiff[T](add: Set[T], remove: Set[T]) extends Diff[Set, T] {
+    override def patch(coll: Set[T]): Set[T] = coll ++ add -- remove
+
+    override def map[U](f: T => U): Diff[Set, U] = AddressDiff(add.map(f), remove.map(f))
+  }
+
+  implicit val _addressDiffable = new AddressDiffable
+
+  private[this] val log = Logger.get(getClass.getName)
+  override def get(req: GetDestination): Stream[Update] = {
+    log.info(s"stream initiated from request: $req")
+    if (!req.scheme.contains("k8s")) {
+      val res = Stream.mk[Update]
+      res.close(GrpcStatus.Unimplemented("Unknown scheme: " + req.scheme.getOrElse("unknown")))
+      return res
+    }
+
+    // This assumes that our dtab begins with the /svc prefix
+    val name = Path(Seq(Buf.Utf8("svc")) ++ req.path.map(Buf.Utf8(_)): _*)
+
+    val stream = Stream.mk[Update]
+    interpreter.bind(Dtab.empty, name).run.changes.respond {
```

Comment:
using respond here and below is a problem because respond creates an observation that we never close.  The really nice thing about VarEventStream is that it handles creating and closing observations properly.  I'd highly recommend trying to massage this into a final Var[Update] using combinators like map and then passing that to VarEventStream than trying to use respond and managing the observations yourself.